docs/advanced/Working_with_git.rst

```diff
@@ -0,0 +1,151 @@
+
+   * **Note:** Commits **do not** technically **change** when we force push the commit ,reset or modify them.But the **hash gets updated**, and the new hashed commit gets tagged to that branch.You can easily find you previous commit on github too just by adding ``/commit/<branch>`` to the repo address.It will show you the **remote git log**.See the hash get changed after rewriting the commit.
+
+#. Sometimes you have **two or more commit on your pull request**,which is usually not desired by maintainers.The solution is to do an interactive rebase and squash the previous **N** commits on the particular branch to which the pull request represents:
+
+   * **Soln:** Please **understand** `interactive rebase <https://help.github.com/articles/about-git-rebase/>`_.Checkout the branch the pull request represents,count the **number[N]** of commits you need to `squash <http://stackoverflow.com/questions/2563632/how-can-i-merge-two-commits-into-one>`_ from the pull requests or ``git log`` on that branch,than ``git rebase -i HEAD~N``.N (**number of commits before**) is usually 2 if you want to **squash 2 commits**.Change the pick to squash.Save the configuration.Than force push the new commit ``git push -f origin <branch>``.
+
+#. If you have **unwanted commits** attached to the pull request or **history is broken** than you need to do an **interactive rebase** :
+
+   * **Soln:** It is better to tell others that you are having such problem as it needs `rewriting history <http://git-scm.com/book/en/v2/Git-Tools-Rewriting-History>`_.Please **understand** `rebasing <https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request>`_ and please see your `logs <http://git-scm.com/book/en/v2/Git-Basics-Viewing-the-Commit-History>`_.The **solution** is an `interactive rebase <https://help.github.com/articles/about-git-rebase/>`_ Command: ``git rebase -i HEAD~N`` (N gives the number of the revisions up in that branch).Try to pick N carefully count the commits as maintainers will ask you to give the finalized result in a single commit.
+
+   * The **interactive rebase** on that branch will show the commands that runs on that branch and finally shows on the pull request.Delete the commits you don't need and keep the ones you need.Squash the needed commits to a single commit.
+
+   * **Note:** **Branches** are technically a **tagging system** to commits that have a hash value ,they are **relative** ,even the master branch.It can operate with commands like squash , pick and many others.The commands are played from **top to bottom** and finally shown in the pull request.See the `options <https://help.github.com/articles/about-git-rebase/>`_.
+
+Resolving merge conflicts
+-------------------------
+Conflicts are essentially two or more commits from different branches which **overlap** cuz they have **different content** on the same revision.Understanding `Merge conflict <https://www.youtube.com/watch?v=zz7NuSCH6II>`_ ,you can manually resolve it with git.
```

Comment:
I think "cuz" is too informal for documentation; we may want to change this to "because".